Simplenote/Classes/SPNoteEditorViewController.h

```diff
@@ -66,9 +58,12 @@
 @property (nonatomic, strong) SPTagView *tagView;
 @property (nonatomic, strong) NSString *searchString;
 
+@property (nonatomic, strong) SPNoteEditorViewControllerSwiftCollaborators *swiftCollaborators;
```

Comment:
I'm not 100% sure about this pattern, because of a few reasons:

1. The name is a bit confusing. It's `swiftCollaborators` but it actually is meant to retain the HistoryViewController
2. The purpose of keeping this reference is to determine if the HistoryViewController is being presented
3. HistoryLoader is used to process the versions, received by the AppDelegate, passed along to the Editor, and then forwarded to the HistoryLoader, which, in turn, is retained by `(..)SwiftCollaborators`, to be consumed by HistoryViewController.

Do you thing this can be simplified a bit?

Reply:
It's not ideal, but what I like in this pattern is that only `SPNoteEditorViewControllerSwiftCollaborators` has to be declared as `@objc` and all fields inside it can be pure swift. It's not only for history, basically any object that counterpart swift extension wants to use, can go there. Classic way here is to put all those objects as a separate properties in the header file `SPNoteEditorViewController.h`.
I can remove `SPNoteEditorViewControllerSwiftCollaborators` and go for a more classic approach 👍 

> The purpose of keeping this reference is to determine if the HistoryViewController is being presented
But that's not a bad approach. I think keeping a weak var to the view controller to determine if it's still alive and on screen is better than having a separate Boolean variable, for example.
The same reference is also used to dismiss `HistoryViewController `

> HistoryLoader is used to process the versions, received by the AppDelegate, passed along to the Editor, and then forwarded to the HistoryLoader, which, in turn, is retained by (..)SwiftCollaborators, to be consumed by HistoryViewController.
With `HistoryLoader` I had to compromise as I didn't want to introduce many changes at once. `SPHistoryLoader` asks bucket to load versions, but can't directly listen for events coming from the bucket. `AppDelegate` receives all the events from Simperium. I went with the weak var to the history loader in the editor because editor was already receiving versions from the app delegate.

Some object has to reference `SPHistoryLoader` to provide versions to it. It can be AppDelegate, but I'm not sure it's better :-) I also don't consider using Notification Center as a way to notify about new versions as I think it makes it too loosely coupled. 

What do you think, @jleandroperez? Maybe you have some ideas how to make it better?

Reply:
I have an idea on how to improve it.
The issue is that we have to forward versions from the AppDelegate to the HistoryLoader and bucket can have only one delegate, which is AppDelegate now (I think technically it's not a delegate, but rather an observer, getting notified about events that happened in a bucket).
What if we make a wrapper around SPBucket which would allow to set several observers. This wrapper can be injected into Editor and HistoryLoader, which would enable them to directly get updates without the need to proxy things from the AppDelegate.

Reply:
@eshurakov 😃  That's exactly what we've got on [the mac here](https://github.com/Automattic/simplenote-macos/blob/develop/Simplenote/VersionsController.swift)

Reply:
👍 I will make changes